code/game/objects/structures/crates_lockers/closets/secure/personal.dm

```diff
@@ -43,13 +43,26 @@
 		to_chat(user, "<span class='warning'>Invalid identification card.</span>")
 		return
 
+	if(opened)
+		to_chat(user, "<span class='notice'>Close the locker first.</span>")
+		return
+
+	if(broken)
+		to_chat(user, "<span class='warning'>The locker appears to be broken.</span>")
+		return
+
+	if(user.loc == src)
+		to_chat(user, "<span class='notice'>You can't reach the lock from inside.</span>")
+		return
+		
 	var/obj/item/card/id/I = W
 	if(!I || !I.registered_name)
 		return
 
 	else if(allowed(user) || !registered_name || (istype(I) && (registered_name == I.registered_name)))
```

Comment:
![image](https://github.com/user-attachments/assets/d46118cb-fa67-4c2f-bf8f-d4eea96a5c97)

Now that you moved this to another places this else if is not necessary.